### ISSUE TITLE
Setting Door-Area-Location(s) is giving an error.

### ISSUE THREAD

**Issue by Jackboittv** [PluginBugs/Issues-CosmeticsCore#188]
### Terms

- [X] I'm using the very latest version of CosmeticsCore and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-CosmeticsCore/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://lonedev.gitbook.io/cosmeticscore/) to know if a solution is already known.
- [X] I already searched on the **Discord** server to know if anyone already provided a solution in the past for the issue.

### Discord tag (optional)

Jackboi#0155

### What happened?

When i use the command: ### /cosmeticscore:cosmeticsconfig config set door-area-locations AND THEN right click anywhere, doesn't matter where, then an error appears.

### Steps to reproduce the issue

When i use the command: ### /cosmeticscore:cosmeticsconfig config set door-area-locations AND THEN right click anywhere, doesn't matter where, then an error appears.

### Server version

![45](https://github.com/PluginBugs/Issues-CosmeticsCore/assets/121448280/7c983d1d-1251-434c-9744-31418b27d2be)

### CosmeticsCore Version

![12](https://github.com/PluginBugs/Issues-CosmeticsCore/assets/121448280/aa0687da-5535-4358-9bdc-c79030b517c4)

### ProtocolLib Version

![11](https://github.com/PluginBugs/Issues-CosmeticsCore/assets/121448280/0c508ac1-4e49-4026-9e9a-3c98298dea0e)

### LoneLibs Version

![111](https://github.com/PluginBugs/Issues-CosmeticsCore/assets/121448280/d7a30c99-c20a-409e-aec4-bdbbe01c8eb1)

### FULL server log

[latest.log](https://github.com/PluginBugs/Issues-CosmeticsCore/files/13487050/latest.log)

### Error (optional)

```shell
[11:52:33] [Server thread/INFO]: JackboiTTV issued server command: /cosmeticscore:cosmeticsconfig config set door-area-locations
[11:52:35] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to CosmeticsCore v1.2.3-r4
com.comphenix.protocol.reflect.FieldAccessException: Field index 0 is out of bounds for length 0
	at com.comphenix.protocol.reflect.FieldAccessException.fromFormat(FieldAccessException.java:49) ~[ProtocolLib.jar:?]
	at com.comphenix.protocol.reflect.StructureModifier.write(StructureModifier.java:316) ~[ProtocolLib.jar:?]
	at dev.lone.cosmeticscore.nms.aI.aM.a(SourceFile:43) ~[CosmeticsCore_1.2.3-r4.jar:?]
	at dev.lone.cosmeticscore.nms.aI.aM.a(SourceFile:27) ~[CosmeticsCore_1.2.3-r4.jar:?]
	at dev.lone.cosmeticscore.main.d.a(SourceFile:188) ~[CosmeticsCore_1.2.3-r4.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:git-Paper-307]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.2.jar:git-Paper-307]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.2.jar:git-Paper-307]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:595) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:537) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:1908) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:37) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.a(ServerboundUseItemOnPacket.java:9) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:53) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1324) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:193) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1301) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1294) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1272) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1160) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.20.2.jar:git-Paper-307]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[11:52:35] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to CosmeticsCore v1.2.3-r4
com.comphenix.protocol.reflect.FieldAccessException: Field index 0 is out of bounds for length 0
	at com.comphenix.protocol.reflect.FieldAccessException.fromFormat(FieldAccessException.java:49) ~[ProtocolLib.jar:?]
	at com.comphenix.protocol.reflect.StructureModifier.write(StructureModifier.java:316) ~[ProtocolLib.jar:?]
	at dev.lone.cosmeticscore.nms.aI.aM.a(SourceFile:43) ~[CosmeticsCore_1.2.3-r4.jar:?]
	at dev.lone.cosmeticscore.nms.aI.aM.a(SourceFile:27) ~[CosmeticsCore_1.2.3-r4.jar:?]
	at dev.lone.cosmeticscore.main.d.a(SourceFile:196) ~[CosmeticsCore_1.2.3-r4.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:81) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:git-Paper-307]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.20.2.jar:git-Paper-307]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.20.2.jar:git-Paper-307]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.20.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:595) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.level.ServerPlayerGameMode.useItemOn(ServerPlayerGameMode.java:537) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItemOn(ServerGamePacketListenerImpl.java:1908) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.handle(ServerboundUseItemOnPacket.java:37) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundUseItemOnPacket.a(ServerboundUseItemOnPacket.java:9) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:53) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1324) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:193) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1301) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1294) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1371) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1156) ~[paper-1.20.2.jar:git-Paper-307]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.20.2.jar:git-Paper-307]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### CosmeticsCore config.yml

```shell
lang: en
extract_default_assets: true
cosmetics:
  max_equippable:
    body: 3
    balloon: 2
  hide_during_combat:
    enabled: true
    combat_duration_ticks: 10000
  fix_invisible_nametag: true
  hide_in_vanish: true
  worlds:
  - '*'
wardrobe_room:
  camera:
    hide_actionbar_texts: true
    rotation:
      enabled: true
    zoom:
      enabled: true
      min: 2
    fade_effect: true
    location:
      world: flat
      x: 205
      y: -58.5
      z: 41
      yaw: 359.85
      pitch: 6.6
  mannequin:
    location:
      world: flat
      x: 204.5
      y: -60
      z: 45.5
      yaw: 180
      pitch: 0
  teleport_area:
    enter:
      world: flat
      pos1:
        x: 167
        y: -61
        z: 42
      pos2:
        x: 167
        y: -57
        z: 37
    exit_location:
      world: flat
      x: 164
      y: -60
      z: 40
      yaw: 98
      pitch: 10
  gui:
    cursor_default: fermented_spider_eye:9000001
    cursor_rotate: fermented_spider_eye:9000002
    cursor_pointer: fermented_spider_eye:9000003
    dyeing_pointer: fermented_spider_eye:9100003
    tutorial_gui: fermented_spider_eye:9100004
    tutorial_gui_checked: fermented_spider_eye:9100005
    slots:
      colors:
        not_owned: 195,147,57
        owned: 56,67,100
        wearing: 57,70,195
        wearing_preview: 195,120,57
        max_amount_reached: 111,111,111
      add_item_in_blank_slots:
        # This is useful only for some expert users that need to do some crazy stuff.
        # Do not enable it if you don't know what you are doing.
        enabled: false
        type: PAPER
        custom_model_data: 1337666
send_creative_hidden_hat_message: true
save:
  # Use only one at a time
  file:
    enabled: true
  mysql:
    enabled: false
    url: jdbc:mysql://my_url_database_69.com:3306/database_name
    username: username
    password: password
    table: cosmeticscore_saved
debug: false
```

**Comment by Jackboittv** [comment]
And yes, the error is preventing me from actually being able to use CosmeticsCore..

**Comment by LoneDev6** [comment]
Thanks a lot for the info, I will fix this as soon as possible.

**Comment by Jackboittv** [comment]
No worries, it did help us both in the end ^^

**Comment by Jackboittv** [comment]
New bug - When left-clicking and then right-click, all works at it should I think. But when I use right-click first, it auto-completes, without having set first position with left-click

**Comment by LoneDev6** [comment]
> New bug - When left-clicking and then right-click, all works at it should I think. But when I use right-click first, it auto-completes, without having set first position with left-click

please make sure to be able to reproduce this and create a new page.
do not report bugs on the same page.


About the screenshots, seems you didn't apply the resourcepack, you won't see anything without the resourcepack applied.

**Comment by mcbaymuratasfdasf** [comment]
Hello, I would like to talk about a bug I found in CosmeticsCore.

When we put on a hat from the wardrobe, when you open the inventory and press the q key on the hat, it seems to disappear and comes back when you left click

Likewise, when we press q on the hat, it disappears. When you put on any Minecraft helmet, this time the hat disappears completely.

I found another bug, when you put on any helmet and go into the wardrobe and put on a hat, the helmet disappears completely.
